RNCalendarReminders.m

```diff
@@ -233,63 +249,142 @@ - (void)addReminderAlarms:(NSString *)eventId alarms:(NSArray *)alarms
     if (!self.isAccessToEventStoreGranted) {
         return;
     }
-    
+
     EKReminder *reminder = (EKReminder *)[self.eventStore calendarItemWithIdentifier:eventId];
     reminder.alarms = [self createReminderAlarms:alarms];
-    
+
     [self saveReminder:reminder];
 }
 
 #pragma mark -
+#pragma mark RecurrenceRules
+
+// Return the EKRecurrenceFrequency value matching a given name.
+-(EKRecurrenceFrequency)frequencyMatchingName:(NSString *)name
+{
+    // Default value
+    EKRecurrenceFrequency recurrence = EKRecurrenceFrequencyDaily;
+    
+    if ([name isEqualToString:@"weekly"])
+    {
+        recurrence = EKRecurrenceFrequencyWeekly;
+    }
+    else if ([name isEqualToString:@"monthly"])
+    {
+        recurrence = EKRecurrenceFrequencyMonthly;
+    }
+    else if ([name isEqualToString:@"yearly"])
+    {
+        recurrence = EKRecurrenceFrequencyYearly;
+    }
+    
+    return recurrence;
+}
+
+
+// Create a recurrence rule
+-(EKRecurrenceRule *)createRecurrenceRule:(NSString *)frequency
+{
+    // Not doing BiWeekly, BiMonthly, etc reminders... at this time.
+    NSUInteger interval = 1;
+    
+    // Check that frequency is valid simple recurrence string.
+    BOOL validFrequency = [@[@"daily", @"weekly", @"monthly", @"yearly"] containsObject:frequency];
+    if (!validFrequency) {
+        return NULL;
+    }
+    
+    // Fetch the EKRecurrenceFrequency value matching frequency, otherwise.
+    EKRecurrenceFrequency recurrenceFrequency = [self frequencyMatchingName:frequency];
+    
+    // Create a recurrence rule using the above recurrenceFrequency and interval
+    EKRecurrenceRule *rule = [[EKRecurrenceRule alloc] initRecurrenceWithFrequency:recurrenceFrequency
+                                                                          interval:interval
+                                                                               end:nil];
+    return rule;
+}
+
+// Return a name for the EKRecurrenceFrequency.
+-(NSString *)nameMatchingFrequency:(EKRecurrenceFrequency)frequency
+{
+    // Default value
+    NSString *name;
+    
+    if (frequency == EKRecurrenceFrequencyDaily)
+    {
+        name = @"daily";
+    }
+    else if (frequency == EKRecurrenceFrequencyWeekly)
+    {
+        name = @"weekly";
+    }
+    else if (frequency == EKRecurrenceFrequencyMonthly)
+    {
+        name = @"monthly";
+    }
+    else if (frequency == EKRecurrenceFrequencyYearly)
+    {
+        name = @"yearly";
+    } else {
+        name = NULL;
```

Comment:
Yes definitely not necessary, forgot to remove that after trying a different approach to validating the recurrence string.